### DISCUSSION TITLE
Source Generate debug is really tough!

### DISCUSSION THREAD

**Discussion by juepiezhongren** [dotnet/roslyn#50123] [General] [answered]
1) i have to do insert code into my generator to make debug possible
```
#if DEBUG
            if (!Debugger.IsAttached) {
                Debugger.Launch();
            }
#endif
```
2) and i have to edit csproj file, rebuild .... to make generator stop debug

these r terrible

**Comment by YairHalberstadt** [comment, marked as answer]
Can you open a discussion about this on dotnet/Roslyn please?

Thanks

**Reply by michael-freidgeim-webjet** [reply]
The similar discussion exists https://github.com/dotnet/roslyn/discussions/50606, but unanswered

**Comment by spydacarnage** [comment]
I suppose a workaround, in the meantime, could be to put above code inside `if (attribute exists in destination assembly)` block instead of an `#IF DEBUG` block.

And then, in the destination assembly, you could have the said attribute inside an `#if DEBUG` block.

Would that work?
